core/src/main/java/org/apache/iceberg/EnvironmentContext.java

```diff
@@ -52,4 +52,13 @@ public static Map<String, String> get() {
   public static void put(String key, String value) {
     PROPERTIES.put(key, value);
   }
+
+  /**
+   * Remove the key from the global properties map.
+   *
+   * @param key The key whose value to remove
+   */
+  public static void remove(String key) {
```

Comment:
maybe the return type should be `String` here and the method would `return PROPERTIES.remove(key)`. That way the test wouldn't need to check `assertThatNoException().isThrownBy(() -> EnvironmentContext.remove("test-key"));` but rather would make sure that the method returned the value on the first remove and null on the second remove

Reply:
Yeah that's nice: 84b2adf41